tests/v2/validation/workloads/workload_test.go

```diff
@@ -292,6 +299,64 @@ func validateDeploymentUpgrade(w *WorkloadTestSuite, namespaceName string, appv1
 	require.Equal(w.T(), expectedContainerCount, countPods)
 }
 
+func (w *WorkloadTestSuite) TestPingPods() {
+	subSession := w.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Infof("Cluster ID %s", w.cluster.ID)
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(w.client, w.cluster.ID)
+	require.NoError(w.T(), err)
+
+	steveClient, err := w.client.Steve.ProxyDownstream(w.cluster.ID)
+	require.NoError(w.T(), err)
+
+	_, err = daemonset.CreateDaemonset(w.client, w.cluster.ID, namespace.Name, 2, "", "", false, false)
+	require.NoError(w.T(), err)
+
+	err = charts.WatchAndWaitDaemonSets(w.client, w.cluster.ID, namespace.Name, metav1.ListOptions{})
+	require.NoError(w.T(), err)
+
+	wc, err := w.client.WranglerContext.DownStreamClusterWranglerContext(w.cluster.ID)
+	require.NoError(w.T(), err)
+
+	createdPods, err := wc.Core.Pod().List(namespace.Name, metav1.ListOptions{})
+	require.NoError(w.T(), err)
+	assert.NotEmpty(w.T(), createdPods)
+
+	pod2Ip := createdPods.Items[1].Status.PodIP
+	pingExecCmd := "ping -c 1 " + pod2Ip
+	nodeRole := "worker"
+	_, stevecluster, err := clusters.GetProvisioningClusterByName(w.client, w.cluster.Name, provisioninginput.Namespace)
+
+	query, err := url.ParseQuery("labelSelector=node-role.kubernetes.io/" + nodeRole + "=true")
+	require.NoError(w.T(), err)
+
+	nodeList, err := steveClient.SteveType("node").List(query)
+	require.NoError(w.T(), err)
+
+	firstMachine := nodeList.Data[0]
+
+	sshUser, err := sshkeys.GetSSHUser(w.client, stevecluster)
+	require.NoError(w.T(), err)
+
+	if sshUser == "" {
+		require.NoError(w.T(), errors.New("sshUser does not exist"))
+	}
+
+	sshNode, err := sshkeys.GetSSHNodeFromMachine(w.client, sshUser, &firstMachine)
+	require.NoError(w.T(), err)
+
+	w.T().Logf("Running ping on [%v]", firstMachine.Name)
+
+	excmdLog, err := sshNode.ExecuteCommand(pingExecCmd)
+	if err != nil && !errors.Is(err, &ssh.ExitMissingError{}) {
```

Comment:
qq, for the case, errors.Is(err, &ssh.ExitMissingError{}), shouldnt we retry in any case if its a transient error?